daft/sql/sql.py

```diff
@@ -185,5 +186,8 @@ def sql(sql: str, catalog: Optional[SQLCatalog] = None, register_globals: bool =
     planning_config = get_context().daft_planning_config
 
     _py_catalog = catalog._catalog
-    _py_logical = _sql(sql, _py_catalog, planning_config)
+
+    sess = daft.current_session()._session
+
+    _py_logical = _sql(sql, _py_catalog, sess, planning_config)
     return DataFrame(LogicalPlanBuilder(_py_logical))
```

Comment:
will need to handle possible different outputs like `sess.sql(..)` since we want to make sure this can execute session statements. I think we should remove `_sql` altogether and could rename `_sql_exec` if you like something different.

Reply:
I've decided to keep the `sql` and `sql_exec` functions as they currently are. While the current design (where `sql` only handles SELECT statements and `sql_exec` handles other statements) is suboptimal, changing this API would significantly expand the scope of this PR. I'll address the API design improvements in a separate PR to keep our changes focused and easier to review.